### ISSUE TITLE
graph:addnode responds before ready causing race condition

### ISSUE THREAD

**Issue by jonnor** [noflo/noflo-runtime-base#114]
`graph:addnode` request sends a response immediately on receiving instead of waiting for the async action to be completed. This causes race conditions when for example followed by `graph:addedge` commands that refer to the added `node`, since the node is not always finished setting up.

This happens easily with fbp-spec on computers with a little bit slow I/O. Or anything else using `fbpClient.graph.send`

Right now the noflo-nodejs runtime crashes when this happens

```
 No process defined for outbound node readSpaces
at Network.addEdge (/home/jon/work/flowhub/bigiot-bridge/node_modules/noflo/lib/Network.js:542:25)
    at processOps (/home/jon/work/flowhub/bigiot-bridge/node_modules/noflo/lib/Network.js:381:31)
    at Graph.<anonymous> (/home/jon/work/flowhub/bigiot-bridge/node_modules/noflo/lib/Network.js:408:18)
    at Graph.emit (events.js:187:15)
    at Graph.addEdge (/home/jon/work/flowhub/bigiot-bridge/node_modules/fbp-graph/lib/Graph.js:587:12)
    at GraphProtocol.addEdge (/home/jon/work/flowhub/bigiot-bridge/node_modules/noflo-runtime-base/protocol/Graph.js:368:18)
    at GraphProtocol.receive (/home/jon/work/flowhub/bigiot-bridge/node_modules/noflo-runtime-base/protocol/Graph.js:40:21)
    at WebSocketRuntime.receive (/home/jon/work/flowhub/bigiot-bridge/node_modules/noflo-runtime-base/Base.js:183:27)
    ...
 +1ms
events.js:167
      throw er; // Unhandled 'error' event
      ^
Error: read ECONNRESET
    at TCP.onread (net.js:659:25)
```

**Comment by jonnor** [comment]
On my system I had to do this haaaacky workaround. Unsure how to fix properly...

```
diff --git a/src/protocol/Graph.coffee b/src/protocol/Graph.coffee
index 23bee47..591387c 100644
--- a/src/protocol/Graph.coffee
+++ b/src/protocol/Graph.coffee
@@ -106,7 +106,9 @@ class GraphProtocol
   subscribeGraph: (id, graph, context) ->
     graph.on 'addNode', (node) =>
       node.graph = id
-      @sendAll 'addnode', node, context
+      setTimeout () =>
+        @sendAll 'addnode', node, context
+      , 500
     graph.on 'removeNode', (node) =>
       nodeData =
         id: node.id
```

**Comment by bergie** [comment]
Should be fixed now with the NoFlo 1.2 features
